tonic/src/transport/server/mod.rs

```diff
@@ -885,3 +1084,27 @@ where
         future::ready(Ok(svc))
     }
 }
+
+// A future which only yields `Poll::Ready` once, and thereafter yields `Poll::Pending`.
+#[pin_project]
```

Comment:
Isn't there something in futures_util we can use for this?

Reply:
Looks like this is exactly `futures-util`s `Fuse` type (except that one uses pin-project-lite).

Reply:
I guess tonic itself doesn't already have a futures-util dependency and does already have a pin-project dependency, so maybe this is reasonable?

Reply:
Yes, this is from `futures-util`, vendored here to avoid the dependency - but either approach is fine. I'll add a comment noting that.

Reply:
avoiding the dependency is the right way 👍